### PR TITLE
Adds a device attribute to "tf_tensor_to_i1" graph ops.

### PR DESCRIPTION
**Pull request by bgogul**
Without the attribute, the device partitioning code fails to introduce TensorTransfer instructions for the operands of tf_tensor_to_i1 even if it was necessary.

Resolves [SR-8426](https://bugs.swift.org/browse/SR-8426).

### CODE REVIEW

**Comment by bgogul** [comment]
@swift-ci please test tensorflow

**Comment by bgogul** [comment]
@swift-ci please test tensorflow

File: lib/SILOptimizer/Mandatory/TFCanonicalizeCFG.cpp [hunk 1]
```diff
@@ -342,6 +342,11 @@ class SingleExitLoopTransformer {
     return SILUndef::get(type, currentFn->getModule());
   }
 
+  /// Create a tf_tensor_to_i1 instruction with the given value as argument.
+  GraphOperationInst *createTFInt1ToBuiltinInt1(SILBuilder &builder,
```
Comment (mhong) [inline lib/SILOptimizer/Mandatory/TFCanonicalizeCFG.cpp]: nit: we don't have the notion of a "tf int1" -- it's more of "tensor handle int1". How about a higher level func name like "createTensorToInt1Inst"?

File: lib/SILOptimizer/Mandatory/TFPartition.cpp [hunk 2]
```diff
@@ -2423,9 +2423,13 @@ void PartitionCloner::visitCondBranchInst(CondBranchInst *inst) {
     assert(eltTy->isBuiltinIntegerType(1) && "expected Tensor<i1>");
 
     auto name = ctx.getIdentifier("tf_tensor_to_i1");
+    SmallVector<GraphOperationAttribute, 1> attributes;
```
Comment (mhong) [inline lib/SILOptimizer/Mandatory/TFPartition.cpp]: is it worth refactoring this and the next call-site into calling createTFInt1ToBuiltinInt1(), and perhaps moving that to TFUtilities?

File: test/TensorFlow/sr8426.swift [hunk 3]
```diff
@@ -0,0 +1,49 @@
+// RUN: %target-swift-frontend -Xllvm -tf-dump-intermediates -Onone -emit-sil -verify %s
```
Comment (mhong) [inline test/TensorFlow/sr8426.swift]: we have a new test case called optimization-disabled. Should we add this test there?

**Review by mhong — APPROVED** [review]
Nice, thanks for the fix!

File: test/TensorFlow/sr8426.swift (hunk 3 above)
Comment (bgogul) [reply test/TensorFlow/sr8426.swift]: good idea. Moved the test there and removed this file.

File: lib/SILOptimizer/Mandatory/TFPartition.cpp (hunk 2 above)
Comment (bgogul) [reply lib/SILOptimizer/Mandatory/TFPartition.cpp]: Done.

**Comment by bgogul** [comment]
@swift-ci please test tensorflow

**Pull request merged**